clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -118,8 +118,10 @@ public ConfigDef define(String name, Type type, Importance importance, String do
         Map<String, Object> values = new HashMap<String, Object>();
         for (ConfigKey key : configKeys.values()) {
             Object value;
-            if (props.containsKey(key.name))
+            if (props.containsKey(key.name)){
                 value = parseType(key.name, props.get(key.name), key.type);
+                if(key.validator!=null) key.validator.ensureValid(key.name,value);
```

Comment:
nit: newline after if condition, also space before and after `!=`, and space after `if`.